crates/node/src/network_server/handler/cluster_ctrl.rs

```diff
@@ -230,6 +230,47 @@ impl ClusterCtrlSvc for ClusterCtrlSvcHandler {
             .map(|_| Response::new(SealAndExtendChainResponse::default()))
             .map_err(|err| Status::internal(err.to_string()))
     }
+
+    async fn find_tail(
```

Comment:
Should we explicitly set a timeout on tail discovery so we can return a more meaningful error to the caller? This probably matters less here since it's the only action being performed, but more important if it's part of another operation like `logs describe --extra`.

Reply:
yeah, definitely a good idea.